### ISSUE TITLE
Cannot reproduce the results in Table 9

### ISSUE THREAD

**Issue by JunhyunB** [thaihungle/SAM#5]
I have some trouble reproducing the results of Table9 (babi QA task).

In our run, the test error for the best model does not fall below 1%.
Actually, the test error is more than 1.4%.

Any suggestion to solve the problem?

```c
validate acc task 1: 1.0
validate acc task 2: 0.989
validate acc task 3: 0.947
validate acc task 4: 1.0
validate acc task 5: 0.984
validate acc task 6: 1.0
validate acc task 7: 0.991
validate acc task 8: 0.996
validate acc task 9: 1.0
validate acc task 10: 1.0
validate acc task 11: 1.0
validate acc task 12: 0.999
validate acc task 13: 0.999
validate acc task 14: 0.998
validate acc task 15: 1.0
validate acc task 16: 0.995
validate acc task 17: 0.999
validate acc task 18: 0.994
validate acc task 19: 0.828
validate acc task 20: 1.0
Test accuracy: 0.986, loss: 0.076
test avg: 0.9859499999999999
```

**Comment by thaihungle** [comment]
How many runs did you try? How many epochs did you set for the training?
Look at your numbers; it seems that only task 19 needs improvement. Typically, if you let the model train for more epochs, it will continue improving.

**Comment by JunhyunB** [comment]
I tried base settings of this repository; 
Run 5 times with maximum 500 training epochs and I found that the results are saturated around 300 epochs.

Should I run more than 500?

**Comment by thaihungle** [comment]
I'll rerun the code to check the result. I only needed to run 500 epochs for my machine to get the result in Table 9. But you can try to run more in the meantime.

**Comment by JunhyunB** [comment]
Can you share the packages as 'requirements.txt' that you used for the experiment?

**Comment by thaihungle** [comment]
I am running the code. If I can reproduce the results with the current package, I'll share it with you.  The old exact environment for the paper is no longer existing.

**Comment by JunhyunB** [comment]
Thank you for your consideration. I'll wait for the results.

**Comment by thaihungle** [comment]
I have pushed the requirements.txt and seed setting for your convenience. 
With torch 1.0.1 and seed 0, I can achieve less than 1% ERR within 54 epochs :
INFO:root:##### EPOCH: 54 #####
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1420/1420 [19:44<00:00,  1.21s/it]
validate acc task 1: 1.0
validate acc task 2: 0.993
validate acc task 3: 0.946
validate acc task 4: 1.0
validate acc task 5: 0.991
validate acc task 6: 0.999
validate acc task 7: 0.987
validate acc task 8: 0.992
validate acc task 9: 1.0
validate acc task 10: 1.0
validate acc task 11: 1.0
validate acc task 12: 0.999
validate acc task 13: 1.0
validate acc task 14: 0.995
validate acc task 15: 1.0
validate acc task 16: 0.975
validate acc task 17: 0.963
validate acc task 18: 1.0
validate acc task 19: 0.985
validate acc task 20: 1.0
saved model...saved_models/model.pt

Test accuracy: 0.993, loss: 0.029
test avg: 0.9928999999999999

**Comment by JunhyunB** [comment]
Thanks a lot!
Did you try experiments on another version of PyTorch since recent GPUs don't support PyTorch 1.0x as I know.

**Comment by thaihungle** [comment]
I tried 1.7.1 but it did not work well. 
By the way, I can run the code with torch 1.0.1 + CUDA 11.3 +  GPU Tesla V100-SXM2.
There is no support issue from my side.

**Comment by JunhyunB** [comment]
Thank you for your work! :)
